src/skills-builder/skills-builder-modal/view-results/RelatedSkillsInteractiveBoxSet.jsx

```diff
@@ -0,0 +1,135 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Card, CardDeck, SelectableBox, Chip, Stack,
+} from '@edx/paragon';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { sendTrackEvent } from '@edx/frontend-platform/analytics';
+import { addCareerInterest } from '../../data/actions';
+import { SkillsBuilderContext } from '../../skills-builder-context';
+import messages from './messages';
+
+/*
+  A variant on the RelatedSkillsSelectableBoxSet that allows for interactive content.
+  Instead of a radio select button in the top, there is a close button.
+  Since the contents can be interactive, this form will allow for selecting skills to
+  drill down for skill-specific recommendations.
+
+  This component needs to be like a CardDeck, but also behave like a radio group.
+*/
+const RelatedSkillsInteractiveBoxSet = ({
+  jobSkillsList, selectedJobTitle, onChange, useInteractiveBoxSet,
+}) => {
+  const { formatMessage } = useIntl();
+  const { state, dispatch } = useContext(SkillsBuilderContext);
+  const { careerInterests } = state;
+
+  const renderTopFiveSkills = (skills) => {
+    const topFiveSkills = skills.sort((a, b) => b.significance - a.significance).slice(0, 5);
+    return (
+      topFiveSkills.map(skill => (
+        <Chip key={skill.external_id} className="chip-max-width">
+          {skill.name}
+        </Chip>
+      ))
+    );
+  };
+
+  const handleCareerInterestSelect = (value) => {
+    if (!careerInterests.includes(value) && careerInterests.length < 3) {
+      dispatch(addCareerInterest(value));
+
+      sendTrackEvent(
+        'edx.skills_builder.career_interest.added',
+        {
+          app_name: 'skills_builder',
+          category: 'skills_builder',
+          learner_data: {
+            career_interest: value,
+          },
+        },
+      );
+    }
+  };
+
+  // eslint-disable-next-line react/prop-types
+  const CardComponent = ({ name, skills }) => (
+    <Card
+      isClickable
+      onClick={handleCareerInterestSelect}
```

Comment:
This doesn't appear to be doing anything when I test in my local environment. This function `handleCareerInterestSelect` is used add to the `careerInterests` array which doesn't change the what the recommendations are in the UI.

I think we need to utilize the `onChange` function here which calls `handleJobTitleChange` from `ViewResults`. Unfortunately, this function is expecting the clicked item to be a control with a `e.target.value` but since it's a card, we'll have to find a different way to pass the value. This value should be the name of the job so that the rest of the function works properly.

Or perhaps we need an entirely new function for this?

Reply:
Correct. This part doesn't actually work. I would love to get your input on how to implement this correctly.

Reply:
Yes, we need an entirely new function for this.